pymediainfo/__init__.py

```diff
@@ -140,20 +140,26 @@ def _parse_xml_data_into_dom(xml_data, encoding_errors="strict"):
     @staticmethod
     def _get_library(library_file=None):
         os_is_nt = os.name in ("nt", "dos", "os2", "ce")
-        if library_file is not None:
+        tryother = None
+        if library_file is None:
             if os_is_nt:
-                return WinDLL(library_file)
+                dll = 'MediaInfo.dll'
+            elif sys.platform == "darwin":
+                dll = 'libmediainfo.0.dylib'
+                tryother = 'libmediainfo.dylib'
             else:
-                return CDLL(library_file)
-        elif os_is_nt:
-            return windll.MediaInfo
-        elif sys.platform == "darwin":
-            try:
-                return CDLL("libmediainfo.0.dylib")
-            except OSError:
-                return CDLL("libmediainfo.dylib")
-        else:
-            return CDLL("libmediainfo.so.0")
+                dll = 'libmediainfo.so.0'
+            library_file = os.path.join(os.path.dirname(__file__), dll)
+            if not os.path.exists(library_file):
+                library_file = dll
+        if os_is_nt:
+            return WinDLL(library_file)
+        try:
+            return CDLL(library_file)
+        except OSError:
+            if tryother is not None:
+                return CDLL(tryother)
```

Comment:
`libmediainfo.dylib` in the package's dir currently won't work, see my previous comment.

Reply:
The package's dir will also hold the 0.dylib file. The tryother is used for linking with system libraries only if 0.dylib cannot be linked with. Is there a problem with that? I do not know about Mach-O linking.

Reply:
I just don't like the `tryother` variable and I find that part of the code hard to understand, what do you think of the way I tried to rewrite it?

Reply:
I do not know which rewrite you are referring to.

Reply:
Damn, it looks like I never submitted my comment :/
I'm going to add this to master:
https://github.com/sbraz/pymediainfo/commit/2d1d81a0d3f8f7ebdc5beec1c7cc11d05bdedc46

Reply:
Let me know if that works for you and I'll merge the branch into master.

Reply:
Works for me.